Edit new information on persist data page (#5944)
circleci/circleci-docs: jekyll/_cci2/persist-data.md

![caching data flow]({{ site.baseurl}}/assets/img/docs/caching-dependencies-overview.png)

**Caches created via the save_cache step are stored for up to 15 days.**

> Caching persists data between the same job in different Workflow builds, allowing you to reuse the data from expensive fetch operations from previous jobs. After an initial job run, future instances will run faster as they will not need to redo the work (provided your cache has not been invalidated).

Reviewer: ```suggestion
Caching persists data between the same job in different builds, allowing you to reuse the data from expensive fetch operations from previous jobs. After an initial job run, future instances will run faster as they will not need to redo the work (provided your cache has not been invalidated).
```

## Using artifacts
{: #using-artifacts }

![artifacts data flow]( {{ site.baseurl}}/assets/img/docs/Diagram-v3-Artifact.png)

**Artifacts are stored for up to 30 days.**

Artifacts are used for longer-term storage of the outputs of your pipelines. For example if you have a Java project, your build will most likely produce a `.jar` file of your code. This code will be validated by your tests. If the whole build/test process passes, then the output of the process (the `.jar`) can be stored as an artifact. The `.jar` file is available to download from our artifacts system long after the workflow that created it has finished.

If your project needs to be packaged, say an Android app where the `.apk` file is uploaded to Google Play, you would likely wish to store it as an artifact. Many users take their artifacts and upload them to a company-wide storage location such as Amazon S3 or Artifactory.

For more information on using artifacts to persist data once a job has completed, see the [Storing Build Artifacts]({{site.baseurl}}/2.0/artifacts/)
guide.

## Managing network and storage use
{: #managing-network-and-storage-use }

> The below information will explain how your network and storage usage is accumulating, and should help you find ways to optimize and implement cost saving measures.

Reviewer 2: ```suggestion
The information below describes how your network and storage usage is accumulating, and should help you find ways to optimize and implement cost saving measures.
```

* Uploading artifacts
* Uploading test results

> To determine which jobs utilize the above actions, you can search for the following commands in your project's `config.yml` file:

Reviewer: I wonder if we should always use `.circleci/_config.yml` to be totally clear when talking about CircleCI config. I am guessing people might have a quite a few `config.yml`s kicking around depending on what other stuff they are using. What do you think?

Author: Yeah that's true, that's a good idea. That probably needs to be its own task as there are many instances throughout the docs that reference simply config.yml

![storage-usage-overage]( {{ site.baseurl }}/assets/img/docs/storage-usage-overage.png)

#### Network
{: #network }
{:.no_toc}

To calculate monthly network costs from your usage, click on the Objects tab to see if your organization has accrued any overages (your network egress). Your overage GB can be multiplied by 420 credits to estimate the total monthly costs.

The GB allotment only applies to outbound traffic from CircleCI. Traffic within CircleCI is unlimited.

![network-usage-overage]( {{ site.baseurl }}/assets/img/docs/network-usage-overage.png)

### How to optimize your storage and network transfer use
{: #how-to-optimize-your-storage-and-network-transfer-use }

There are several common ways that your configuration can be optimized to ensure you are getting the most out of your storage and network usage.

For example, when looking for opportunities to reduce data usage, consider whether specific usage is providing enough value to be kept.

In the cases of caches and workspaces this can be quite easy to compare - does the developer or compute time-saving from the cache outweigh the cost of the download and upload?

> Please see below for examples of storage and network optimization opportunities through reducing artifact, cache, and workspace traffic.

Reviewer: We try not to say "please". Maybe: "The sections below show examples of storage ..."

#### Check which artifacts are being uploaded
{: #check-which-artifacts-are-being-uploaded }

Often we see that the `store_artifacts` step is being used on a large directory when only a few files are really needed, so a simple action you can take is to check which artifacts are being uploaded and why.

If you are using parallelism in your jobs, it could be that each parallel task is uploading an identical artifact. You can use the `CIRCLE_NODE_INDEX` environment variable in a run step to change the behavior of scripts depending on the parallel task run.

#### Uploading large artifacts
{: #uploading-large-artifacts }

Artifacts that are text can be compressed at very little cost.

If you are uploading images/videos of UI tests, filter out and upload only failing tests. Many organizations upload all of the images from their UI tests, many of which will go unused.

> If your pipelines build a binary, uberjar, consider if these are necessary for every commit? You may wish to only upload artifacts on failure / success, or perhaps only on a single branch, using a filter.

Reviewer: Not quite sure of the meaning here but us uberjar an example? Or should this be "binary or uberjar, consider" I wonder?

If you must upload a large artifact you can upload them to your own bucket at no cost.

#### Caching unused or superfluous dependencies
{: #caching-unused-or-superfluous-dependencies }

Depending on what language and package management system you are using, you may be able to leverage tools that clear or “prune” unnecessary dependencies.

> For example, the node-prune package removes unnecessary files (markdown, typescript files, etc.) from `node_modules`.

Reviewer 2: ```suggestion
For example, the `node-prune` package removes unnecessary files (markdown, typescript files, etc.) from `node_modules`.
```

```
{% endraw %}

> Notice in the above example that best practices are not being followed. `brew-{{ epoch }}` will change every build; causing an upload every time even if the value has not changed. This will eventually cost you money, and never save you any time. Instead pick a cache `key` like the following:

Reviewer: ```suggestion
Notice in the above example that best practices are not being followed. `brew-{{ epoch }}` will change every build causing an upload every time even if the value has not changed. This will eventually cost you money, and never save you any time. Instead pick a cache `key` like the following:
```
